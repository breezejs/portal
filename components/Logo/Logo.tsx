import React, {memo, ReactNode} from 'react';
import styled from 'styled-components';
import Typography from '../Typography';

const StyledTypography = styled(Typography)`
  margin-bottom: 0;
`;

interface ILogoProps {
  children: ReactNode;
}

function Logo ({children}: ILogoProps) {
  return (
    <StyledTypography component='h1' variant='h4'>{children}</StyledTypography>
  );
}

export default memo(Logo);
