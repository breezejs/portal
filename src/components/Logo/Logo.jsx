import PropTypes from 'prop-types';
import React, {memo} from 'react';
import styled from 'styled-components';
import Typography from '../Typography';

const StyledTypography = styled(Typography)`
  margin-bottom: 0;
`;

function Logo ({children}) {
  return (
    <StyledTypography component='h1' variant='h4'>{children}</StyledTypography>
  );
}

Logo.propTypes = {
  children: PropTypes.string.isRequired
};

export default memo(Logo);
