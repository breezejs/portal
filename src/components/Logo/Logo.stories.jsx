import {text, withKnobs} from '@storybook/addon-knobs';
import {storiesOf} from '@storybook/react';
import React from 'react';
import GlobalStyles from '../GlobalStyles';
import Logo from './Logo';

const stories = storiesOf('Logo', module);

stories
  .addDecorator(withKnobs)
  .addDecorator((storyFn) => (
    <div>
      <GlobalStyles />
      {storyFn()}
    </div>
  ))
  .add('with brand text', () => (
    <Logo>{text('Brand', 'BreezeJS')}</Logo>
  ));
