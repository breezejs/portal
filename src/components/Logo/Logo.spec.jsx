import {mount, shallow} from 'enzyme';
import React from 'react';
import Logo from './Logo';

describe('Logo', () => {
  it('should render', () => {
    const result = shallow(
      <Logo>BreezeJS</Logo>
    );

    expect(result).toMatchSnapshot();
  });

  it('should contain h1 element with the correct styles', () => {
    const result = mount(
      <Logo>BreezeJS</Logo>
    );

    expect(result.find('h1')).toMatchSnapshot();
  });
});
