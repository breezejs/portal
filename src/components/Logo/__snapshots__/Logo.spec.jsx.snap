// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should contain h1 element with the correct styles 1`] = `
.c0 {
  margin-bottom: 0;
}

.c1 {
  font-size: 1.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<h1
  className="c0 c1"
>
  BreezeJS
</h1>
`;

exports[`Logo should render 1`] = `
<Styled(Component)
  component="h1"
  variant="h4"
>
  BreezeJS
</Styled(Component)>
`;
